File: Stack/BalancedBrackets.cpp
#include <bits/stdc++.h>
using namespace std;
bool balancedBrackets(string s)
{
    stack<char> stk;

    for (int i = 0; i < s.length(); i++)
    {
        char check;
        if (s[i] == '{' || s[i] == '[' || s[i] == '(')
        {
            stk.push(s[i]);
            continue;
        }
        if (stk.empty())
        {
            return false;
        }

        switch (s[i])
        {
        case '}':
            check = stk.top();
            stk.pop();
            if (check == '[' || check == '(')
                return false;
            break;
        case ']':
            check = stk.top();
            stk.pop();
            if (check == '{' || check == '(')
                return false;
            break;
        case ')':
            check = stk.top();
            stk.pop();
            if (check == '[' || check == '{')
                return false;
            break;
        }
    }
    return (stk.empty());
}
int main()
{
    string s;
    cin >> s;
    if (balancedBrackets(s))
    {
        cout << "Balanced";
    }
    else
    {
        cout << "Not Balanced";
    }
    return 0;
}

File: Stack/BalancedBrackets.java
package Stack;

import java.util.*;
import java.util.ArrayDeque;
import java.util.Deque;

public class BalancedBrackets {
    static boolean balancedbrackets(String s) {
        Deque<Character> stack = new ArrayDeque<Character>();

        for (int i = 0; i < s.length(); i++) {
            char x = s.charAt(i);
            if (x == '{' || x == '(' || x == '[') {
                stack.push(x);
                continue;
            }
            if (stack.isEmpty()) {
                return false;
            }
            char check;
            switch (x) {
                case '}':
                    check = stack.pop();
                    if (check == '[' || check == '(')
                        return false;
                    break;
                case ']':
                    check = stack.pop();
                    if (check == '{' || check == '(')
                        return false;
                    break;
                case ')':
                    check = stack.pop();
                    if (check == '[' || check == '{')
                        return false;
                    break;

            }
        }
        return (stack.isEmpty());
    }

    public static void main(String[] args) {

        Scanner scan = new Scanner(System.in);
        String s = scan.next();

        if (balancedbrackets(s)) {
            System.out.println("Balanced");
        } else {
            System.out.println("Not Balanced");
        }

    }
}